test/e2e/e2e_suite_test.go

```diff
@@ -1,13 +1,109 @@
 package e2e_test
 
 import (
+	"context"
 	"testing"
+	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	c client.Client
 )
 
 func TestE2E(t *testing.T) {
 	RegisterFailHandler(Fail)
+	SetDefaultEventuallyTimeout(1 * time.Minute)
+	SetDefaultEventuallyPollingInterval(1 * time.Second)
 	RunSpecs(t, "E2E Suite")
 }
+
+var _ = BeforeSuite(func() {
+	config := ctrl.GetConfigOrDie()
+
+	scheme := runtime.NewScheme()
+	err := v1alpha1.AddToScheme(scheme)
+	Expect(err).To(BeNil())
+
+	c, err = client.New(config, client.Options{Scheme: scheme})
+	Expect(err).To(BeNil())
+})
+
+var _ = Describe("k8s provisioner", func() {
+	var (
+		bundle *v1alpha1.Bundle
+	)
+	BeforeEach(func() {
+		By("creating the testing Bundle resource")
+		bundle = &v1alpha1.Bundle{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "olm-crds",
+			},
+			Spec: v1alpha1.BundleSpec{
+				ProvisionerClassName: "core.rukpak.io/plain",
+				Image:                "quay.io/tflannag/olm-plain-bundle:olm-crds-v0.20.0",
+			},
+		}
+		err := c.Create(context.Background(), bundle)
+		Expect(err).To(BeNil())
+	})
+	AfterEach(func() {
+		By("deleting the testing Bundle resource")
+		Eventually(func() error {
+			return c.Delete(context.Background(), bundle)
+		}).Should(Succeed())
```

Comment:
Curious why we need Eventually here. Do we expect `Delete` calls to fail temporarily?